llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -9222,6 +9223,18 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
                                             OpTE->Scalars.size());
     }
 
+    // Calculate the cost difference of propagating a vector vs series of
+    // scalars across blocks. This may be nonzero in the case of illegal
+    // vectors.
```

Comment:
The comment talks about illegal vector types but the code affects only legal vector types